clang/lib/Sema/SemaTemplateInstantiateDecl.cpp

```diff
@@ -629,15 +634,15 @@ static void instantiateSYCLIntelLoopFuseAttr(
 }
 
 template <typename AttrName>
-static void instantiateIntelSYCLFunctionAttr(
-    Sema &S, const MultiLevelTemplateArgumentList &TemplateArgs,
-    const AttrName *Attr, Decl *New) {
+static void
+instantiateFunctionAttr(Sema &S,
```

Comment:
Same concern about the rename here, tbh. This whole file is about template instantiation, and this function definitely does not instantiate generic function attributes (it's quite specific to intel's attributes).

Reply:
Ok. With that in mind, would you prefer if I created separate functions i.e. `instantiatexyzattribute ` for each attribute, I am going to modify now? . It would result in a fair amount of code duplication but I believe this is how clang handles most attributes

Reply:
I'd say hold off on that for now -- we may want to do it later, but for right now I think the logic is reasonable and the name is reasonable. Once we figure out a better way to split `addIntelSYCLSingleArgFunctionAttr()` up, we'll probably have a better idea of what to do here for template instantiation.